docs(docker): rework docker installation page

akantcheff opened the pull request.

akantcheff committed:
docs(docker): rework docker installation page

akantcheff committed:
REVERT-ME: link to chore/add-asciidoctor-tabs

rbioteau commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,56 @@
+# Example on how to use Postgres 15 with Bonita in a cluster mode
+services:
+  bonita-db:
+    image: bonitasoft/bonita-postgres:15.4
```
15.4 does not exist

akantcheff replied:
https://github.com/Bonitasoft-Community/bonita-database-docker/pull/12

rbioteau commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,56 @@
+        condition: service_healthy
+
+  traefik:
+    image: traefik:v2.9
```
```suggestion
    image: traefik:v2.10.5
```

rbioteau commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,56 @@
+    #volumes:
+    #  - ~/my/test/dumps:/opt/bonita/dump
+
+  bonita:
```
in the case of an sca, the service should have the app name

akantcheff replied:
I agree but this is a generic example

akantcheff committed:
fix(tabs): remove comments that broke tabs layout

akantcheff commented on .github/workflows/publish-pr-preview.yml:
```diff
@@ -20,3 +20,4 @@ jobs:
           surge-token: ${{ secrets.SURGE_TOKEN_DOC }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
           component-name: bonita
+          doc-site-branch: 'chore/add-asciidoctor-tabs'
```
Do not forget to revert this

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Commit:
Update docker-compose.yml

akantcheff commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,59 @@
+  # my-application:
+  #  image: my-application-[my built environment]:[my aplication's semantic version]
+  bonita:
+    image: bonitasoft.jfrog.io/docker/bonita-subscription:[{bonitaVersion} for the latest maintenance version or a specific maintenance version such as {bonitaVersion}-u0]
```
@vhemery I don't understand why did you put this sentence here 🤔

vhemery replied:
Because I didn't understand the "pass:a" ad was mistaken by incorrect files with where "pass:" was missing (displaying "a2023.1").

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Docker compose deployment: Clarify version and give example for built Applications

Commit:
Update docker-compose.yml

Commit:
Update docker-compose.yml

Commit:
Update docker-compose.yml

Commit:
Update docker-compose.yml

akantcheff commented:
@vhemery can you commit all changes in one batch please to easily see changes and to prevent rebuilding the PR for each commits

vhemery commented:
> @vhemery can you commit all changes in one batch please to easily see changes and to prevent rebuilding the PR for each commits

I wish I could with Github edition... (on second though, I probably could have done it with a PR to the PR branch)
All done now.

tbouffard commented:
Put in draft as this PR requires
- https://github.com/bonitasoft/bonita-documentation-site/pull/636 
- revert the GH workflow changes to use the master branch of bonita-documentation-site

Commit:
Revert the GH workflow changes to use the master branch of bonita-documentation-site

rbioteau commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+# Example on how to use Postgres 15 with Bonita in a cluster mode
+services:
+  bonita-db:
+    image: bonitasoft/bonita-postgres:15.4
```
```suggestion
    image: bonitasoft/bonita-postgres:latest
```
Use latest tag to simplify maintenance

rbioteau commented on modules/runtime/examples/docker/community/mysql/docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+# Example on how to use MySQL 8.0 with a Community edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-mysql:8.0.33
```
```suggestion
    image: bonitasoft/bonita-mysql:latest
```

rbioteau commented on modules/runtime/examples/docker/community/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,35 @@
+# Example on how to use Postgres 15 with a Community edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-postgres:15.4
```
```suggestion
    image: bonitasoft/bonita-postgres:latest
```

rbioteau commented on modules/runtime/examples/docker/community/sqlserver/docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+# Example on how to use SQL Server 2022 with a Community edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-sqlserver:2022-CU4
```
```suggestion
    image: bonitasoft/bonita-sqlserver:latest
```

rbioteau commented on modules/runtime/examples/docker/subscription/mysql/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+# Example on how to use MySQL 8.0 with a Subscription edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-mysql:8.0.33
```
```suggestion
    image: bonitasoft/bonita-mysql:latest
```

rbioteau commented on modules/runtime/examples/docker/subscription/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,41 @@
+# Example on how to use Postgres 15 with a Subscription edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-postgres:15.4
```
```suggestion
    image: bonitasoft/bonita-postgres:latest
```

rbioteau commented on modules/runtime/examples/docker/subscription/sqlserver/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+# Example on how to use SQL Server 2022 with a Subscription edition of Bonita
+services:
+  bonita-db:
+    image: bonitasoft/bonita-sqlserver:2022-CU4
```
```suggestion
    image: bonitasoft/bonita-sqlserver:latest
```

Commit:
Update modules/runtime/examples/docker/cluster/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/mysql/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/postgres/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/sqlserver/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/postgres/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/sqlserver/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/mysql/docker-compose.yml

Co-authored-by: Romain Bioteau <romain.bioteau@bonitasoft.com>

Commit:
Merge branch '2023.2' into doc/docker-installation

educhastenier commented on modules/runtime/examples/docker/cluster/docker-compose.yml:
```diff
@@ -0,0 +1,60 @@
+# Example on how to use Postgres 15 with Bonita in a cluster mode
```
```suggestion
# Example on how to use Postgres (latest available version) with Bonita in a cluster mode
```

educhastenier commented on modules/runtime/examples/docker/community/mysql/docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+# Example on how to use MySQL 8.0 with a Community edition of Bonita
```
```suggestion
# Example on how to use MySQL (latest available version) with a Community edition of Bonita
```

educhastenier commented on modules/runtime/examples/docker/community/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,35 @@
+# Example on how to use Postgres 15 with a Community edition of Bonita
```
```suggestion
# Example on how to use Postgres (latest available version) with a Community edition of Bonita
```

educhastenier commented on modules/runtime/examples/docker/community/sqlserver/docker-compose.yml:
```diff
@@ -0,0 +1,32 @@
+# Example on how to use SQL Server 2022 with a Community edition of Bonita
```
```suggestion
# Example on how to use SQL Server (latest available version) with a Community edition of Bonita
```

educhastenier commented on modules/runtime/examples/docker/subscription/mysql/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+# Example on how to use MySQL 8.0 with a Subscription edition of Bonita
```
```suggestion
# Example on how to use MySQL (latest available version) with a Subscription edition of Bonita
```

educhastenier commented on modules/runtime/examples/docker/subscription/postgres/docker-compose.yml:
```diff
@@ -0,0 +1,41 @@
+# Example on how to use Postgres 15 with a Subscription edition of Bonita
```
```suggestion
# Example on how to use Postgres (latest available version) with a Subscription edition of Bonita
```

educhastenier commented on modules/runtime/examples/docker/subscription/sqlserver/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+# Example on how to use SQL Server 2022 with a Subscription edition of Bonita
```
```suggestion
# Example on how to use SQL Server (latest available version) with a Subscription edition of Bonita
```

educhastenier reviewed (commented):
Partial review...

Commit:
Update modules/runtime/examples/docker/cluster/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/mysql/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/postgres/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/sqlserver/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/mysql/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/subscription/sqlserver/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>

Commit:
Update modules/runtime/examples/docker/community/postgres/docker-compose.yml

Co-authored-by: Emmanuel Duchastenier <emmanuel.duchastenier@bonitasoft.com>
